OpenTrons/opentrons-api
# feat(robot-server): Load endpoint supplied deck configuration into protocol engine

# Overview
Robot server side of the changes resolving deck configuration support between server and PE.

The process of putting a deck configuration list of Cutout fixtures now includes injecting the list of fixtures into the protocol engine. This occurs twice, first when the protocol engine is created and again bounded to the play action to ensure the most up to deck configuration is utilized. 

# Test Plan
Execute the following series of actions on a Flex:

- PUT a /deck_configuration to the robot, with a full list of deck fixtures including a "stagingAreaRightSlot" in B3 and a "stagingAreaSlotWithWasteChuteRightAdapterNoCover" in D3.

- Load the team internally built addressable areas JSON protocol which will load an armadillo plate to C1, have the gripper move it to B4, then into the Waste Chute.

- Ensure the JSON Protocol passes analysis.

- Run the protocol, ensure all events occur as described end-to-end and the gantry moves to appropriate areas of the deck as expected, and that the protocol ends gracefully.
